frontend/src/features/admin-form/responses/ResponsesPage/storage/useDecryptionWorkers.ts

```diff
@@ -27,18 +27,21 @@ export type DownloadEncryptedParams = EncryptedResponsesStreamParams & {
 const useDecryptionWorkers = () => {
   const [workers, setWorkers] = useState<CleanableDecryptionWorkerApi[]>([])
   const abortControllerRef = useRef(new AbortController())
+
   const { data: adminForm } = useAdminForm()
 
   const { refetch } = useFormResponsesCount()
 
   useEffect(() => {
-    const abortController = abortControllerRef.current
+    return () => killWorkers(workers)
+  }, [workers])
 
+  useEffect(() => {
```

Comment:
why use two separate `useEffect`s?

Reply:
one is dependent on workers, the other isn't haha.